File: Sem3/Advanced-Programming/Python/Stack.py
class Stack:
    def __init__(self):
        self.stack = []

    def push(self, item):
        """Add an item to the top of the stack."""
        self.stack.append(item)

    def pop(self):
        """Remove and return the top item from the stack. Raises an exception if the stack is empty."""
        if self.is_empty():
            raise IndexError("Pop from an empty stack")
        return self.stack.pop()

    def peek(self):
        """Return the top item of the stack without removing it. Raises an exception if the stack is empty."""
        if self.is_empty():
            raise IndexError("Peek from an empty stack")
        return self.stack[-1]

    def is_empty(self):
        """Return True if the stack is empty, False otherwise."""
        return len(self.stack) == 0

    def size(self):
        """Return the number of items in the stack."""
        return len(self.stack)

# Create a stack
s = Stack()

# Push elements onto the stack
s.push(14)
s.push(03)
s.push(05)

# Get the top element
print("Top element is:", s.peek())  # Output: Top element is: 15

# Pop the top element
print("Popped element is:", s.pop())  # Output: Popped element is: 15

# Check if the stack is empty
print("Is stack empty?", s.is_empty())  # Output: Is stack empty? False

# Get the current size of the stack
print("Current size of the stack:", s.size())  # Output: Current size of the stack: 2

File: Sem4/DSA/Lab/Stack.c
#include <stdio.h>
#define MAX 10
int stack[MAX];
int top = -1;
void push(int item) {
    if(top >= MAX-1) {
        printf("\nStack Overflow");
        return;
    }
    stack[++top] = item;
}
int pop() {
    if(top < 0) {
        printf("\nStack Underflow");
        return -1;
    }
    return stack[top--];
}
int peek() {
    if(top < 0) {
        printf("\nStack is Empty");
        return -1;
    }
    return stack[top];
}
void display() {
    if(top < 0) {
        printf("\nStack is Empty");
        return;
    }
    printf("\nStack elements are:\n");
    for(int i = top; i >= 0; i--)
        printf("%d\n", stack[i]);
}
int main() {
push(10);
push(20);
push(30);
push(40);
push(50);
display();
    
pop();
pop();
peek();
display(); 
return 0;     
}
